src/coreclr/System.Private.CoreLib/src/System/Reflection/RuntimeConstructorInfo.cs

```diff
@@ -95,7 +105,27 @@ internal INVOCATION_FLAGS InvocationFlags
         internal override bool CacheEquals(object? o) =>
             o is RuntimeConstructorInfo m && m.m_handle == m_handle;
 
-        private Signature Signature => m_signature ??= new Signature(this, m_declaringType);
+        private Signature Signature
+        {
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            get
+            {
+                Signature? signature = m_signature;
```

Comment:
Technically this should also be a Volatile.Read, but in practice it won't matter.

Reply:
Really? Isn't the write being published last the only thing that matters, since if a reader sees this field as non-null they're already guaranteed the object's properties have been published beforehand?

I can make the field volatile if it's technically the correct thing to do, but I want to make sure I understand this first.

Reply:
_Technically_ without it additional reads could be introduced, e.g.
```C#
Signature s = m_signature;
if (m_signature is null) { ...}
return s;
```

Reply:
Huh, interesting. I knew the JIT often optimizes multiple ldfld instructions into a single true memory read, but I didn't realize the opposite direction (one ldfld resulting in multiple reads) was also legal.

Thanks for the info!